independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/SwitchToEventLoopHandler.java

```diff
@@ -0,0 +1,35 @@
+package org.jboss.resteasy.reactive.server.handlers;
+
+import java.util.concurrent.Executor;
+import java.util.function.Supplier;
+
+import org.jboss.resteasy.reactive.server.core.BlockingOperationSupport;
+import org.jboss.resteasy.reactive.server.core.ResteasyReactiveRequestContext;
+import org.jboss.resteasy.reactive.server.spi.ServerRestHandler;
+
+/**
+ * A special handler whose only purpose is to switch execution from a worker thread to the event loop
+ * This can be needed when for example a large payload is read but the JAX-RS Resource method
+ * is intended to be run on the event loop.
+ */
+public class SwitchToEventLoopHandler implements ServerRestHandler {
```

Comment:
I don't see why we need this handler when we're already resuming the event loop here: https://github.com/quarkusio/quarkus/pull/31613/files#diff-7ae4aff73637bd0d5f0efd30d5c595a8881dec687c6ad2d626d300800c165ccaR86

Reply:
Oops! 
The change in InputHandler has not effect! It was left over from a previous attempt to fix the problem